core/capabilities/remote/target/caller.go

```diff
@@ -0,0 +1,141 @@
+package target
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	commoncap "github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/remote/target/request"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/remote/types"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+type callerRequest interface {
+	OnMessage(ctx context.Context, msg *types.MessageBody) error
+	ResponseChan() <-chan commoncap.CapabilityResponse
+	Expired() bool
+	Cancel(reason string)
+}
+
+// caller/Receiver are shims translating between capability API calls and network messages
```

Comment:
can you expand this comment with godoc links to the capability API and network message definitions as well an explanation about why we need the shim?